mobile-app/app/screens/TransactionAuthorization/PasscodePrompt.tsx

```diff
@@ -33,157 +36,128 @@ interface PasscodePromptProps {
-                    <ThemedText
-                      testID='txn_authorization_description'
-                      dark={tailwind('text-gray-400')}
-                      light={tailwind('text-gray-500')}
-                      style={tailwind('text-sm text-center')}
-                    >
-                      {props.transaction.description}
-                    </ThemedText>
-                  )
-                }
-              </View>
-
-              {
-                props.status === TransactionStatus.PIN && (
-                  <PinTextInput
-                    cellCount={props.pinLength}
-                    onChange={(pin) => {
-                      props.onPinInput(pin)
-                    }}
-                    testID='pin_authorize'
-                    value={props.pin}
-                  />
+                  {props.successMessage}
+                </ThemedTextV2>
+            }
+          {// show loading message
+              props.status === TransactionStatus.SIGNING &&
+                <ThemedTextV2
+                  testID='txn_authorization_message'
+                  dark={tailwind('text-mono-dark-v2-700')}
+                  light={tailwind('text-mono-light-v2-700')}
+                  style={tailwind('px-8 text-sm text-center font-normal-v2')}
+                >
+                  {translate('screens/UnlockWallet', props.loadingMessage)}
+                </ThemedTextV2>
+            }
+          {// show default message
+              (!props.isRetry && props.status !== TransactionStatus.AUTHORIZED && props.status !== TransactionStatus.SIGNING) &&
+                <ThemedTextV2
+                  testID='txn_authorization_message'
+                  dark={tailwind('text-mono-dark-v2-700')}
+                  light={tailwind('text-mono-light-v2-700')}
+                  style={tailwind('text-sm text-center font-normal-v2')}
+                >
+                  {translate('screens/UnlockWallet', props.message)}
+                </ThemedTextV2>
+            }
+          {// hide description when passcode is incorrect or verified.
+            (props.transaction?.description !== undefined && !props.isRetry && props.status !== TransactionStatus.AUTHORIZED) &&
+              (
+                <ThemedTextV2
+                  testID='txn_authorization_description'
+                  dark={tailwind('text-mono-dark-v2-700')}
+                  light={tailwind('text-mono-light-v2-700')}
+                  style={tailwind('text-sm text-center font-normal-v2')}
+                >
+                  {props.transaction.description}
+                </ThemedTextV2>
+              )
+            }
+          {// upon retry: show remaining attempt allowed
+              (props.isRetry && props.attemptsRemaining !== undefined && props.attemptsRemaining !== props.maxPasscodeAttempt && props.status !== TransactionStatus.SIGNING) &&
+                (
+                  <Text
```

Comment:
to be consistent, let's use `<ThemedText/>` to be sure we have handled theme styling